pre-processing.py

```diff
@@ -0,0 +1,116 @@
+# if kernel crashes, make sure pywin32 and pipywin32 are installed.
+# Followed instructions here: https://github.com/jupyter/notebook/issues/4909
+# import win32api
+
+import pandas as pd
+import numpy as np
+import plotly.graph_objects as go
+import plotly.express as px
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+from dash.dependencies import Input, Output
+import dash_bootstrap_components as dbc
+import microdf as mdf
+import os
+import us
+
+# Import data from Ipums
+person_raw = pd.read_csv("cps_00041.csv.gz")
+
+# Create copy and lower column names
+person = person_raw.copy(deep=True)
+person.columns = person.columns.str.lower()
+person.asecwt /= 3
+
+# Crate booleans for demographics
+person["adult"] = person.age > 17
+person["child"] = person.age < 18
+
+person["black"] = person.race == 200
+person["white_non_hispanic"] = (person.race == 100) & (person.hispan == 0)
+person["hispanic"] = (person.hispan > 1) & person.hispan < 700
+person["pwd"] = person.diffany == 2
+person["non_citizen"] = person.citizen == 5
+person["non_citizen_child"] = (person.citizen == 5) & (person.age < 18)
+person["non_citizen_adult"] = (person.citizen == 5) & (person.age > 17)
+
+# Remove NIUs
+person["taxinc"].replace({9999999: 0}, inplace=True)
+person["adjginc"].replace({99999999: 0}, inplace=True)
+person["incss"].replace({999999: 0}, inplace=True)
+person["incssi"].replace({999999: 0}, inplace=True)
+person["incunemp"].replace({99999: 0}, inplace=True)
+person["incunemp"].replace({999999: 0}, inplace=True)
+person["ctccrd"].replace({999999: 0}, inplace=True)
+person["actccrd"].replace({99999: 0}, inplace=True)
+person["eitcred"].replace({9999: 0}, inplace=True)
+person["fica"].replace({99999: 0}, inplace=True)
+person["fedtaxac"].replace({99999999: 0}, inplace=True)
+person["stataxac"].replace({9999999: 0}, inplace=True)
+
+# Change fip codes to state names
+person["statefip"]=person["statefip"].astype(str)
+person["statefip"]=person["statefip"].apply(lambda x: us.states.lookup(x))
+person["statefip"]=person["statefip"].astype('str')
```

Comment:
```suggestion
person["statefip"]=person["statefip"].astype(str)
```
for consistency with prior line